stack_data/core.py

```diff
@@ -669,6 +700,7 @@ def lines(self) -> List[Union[Line, LineGap]]:
         if not pieces:
             return []
 
+        common_markers_block_indent = self._markers_block_indent()
```

Comment:
> not everyone using `lines` is interested in highlighting the executing node, hence my request to only do this computation in executing_node_ranges.

> I suggest adding a private cached property to FrameInfo and using it from Line.executing_node_ranges.

Reply:
When I try to do this, I get a RecursionError.

As I understand it, here's why:

FrameInfo._markers_block_indent() creates **temporary** Line objects for its internal computation; these are not the cached FrameInfo.lines; those are properly obtained only after FrameInfo._markers_block_indent() has been calculated. 

I do need these temporary Line objects to figure out what common indentation should be used.

These temporary Line objects are created before FrameInfo._markers_block_indent() has finished. If I move the calculation in Line.executing_node_range, each temporary Line object will in turn call Frame.frame_info._markers_block_indent, which will create new temporary Line objects, etc.

Reply:
You don't need temporary line objects - see how `leading_indent` is computed. But that's not really the point. Set the default `Line._markers_block_indent` to `None`, and check for that to decide if it needs to be computed.

Reply:
This common marker_block_indent is used to compute Line.range_from_node; unlike Line.leading_indent, this is a NamedTuple (RangeInLine) which cannot be modified once computed; so, once a Line object has been created, this property can no longer be modified.  Even if it could, it still would not really help me, since an other relevant cached property, Line.executing_node_ranges, could not be changed.

I cannot compute a marker_block_indent based on a single Line; I need a block of lines to do so. 

So, I create a temporary version of all the lines that belong to an executing piece of a frame, using the default value for range.start for each line.  Each of these temporary lines has a (possibly incorrect) value of its cached property Line.executing_node_ranges, since I do not know the common indentation value to use at this point.

Then, I use these temporary Line objects to compute the "true" lines that belong to the FrameInfo, using this commmon indentation value.

In summary: Lines are, for practical purpose, immutable objects. To compute their common indentation, I need to use many of them (belonging to a common indented block), to determine their common indentation value. Since I cannot mutate these Line objects, I must recreate some new ones with the appropriate values.

To avoid creating temporary lines, at the very least, I would have to replace the NamedTuple RangeInLine by a mutable class, so that its `start` attribute could be changed.  Even then,